argopy/xarray.py

```diff
@@ -547,3 +583,68 @@ def profile2point(self):
         ds.argo._add_history('Transformed with profile2point')
         ds.argo._type = 'point'
         return ds
+
+    def interp_std_levels(self, std_lev):
+        """ Returns a new dataset interpolated to new inputs levels """
+
+        if self._type != 'profile':
+            raise InvalidDatasetStructure(
+                "Method only available to a collection of profiles")
+
+        # if(self._mode != 'standard'):
+        #    raise InvalidDatasetStructure(
+        #        "Method only available for the standard mode yet")
+
+        if (type(std_lev) is np.ndarray) | (type(std_lev) is list):
+            std_lev = np.array(std_lev)
+            if (np.any(sorted(std_lev) != std_lev)) | (np.any(std_lev < 0)):
+                raise ValueError(
+                    'Standard levels must a numpy array of positive and sorted values')
+        else:
+            raise ValueError('Standard levels must a numpy array or a list')
+
+        ds = self._obj
+
+        # Filtering on pressure levels to avoid extrapolation, this is open to discussion...
+        # i1 = (ds['PRES'].min('N_LEVELS') <= std_lev[0]) & (
+        #    ds['PRES'].max('N_LEVELS') >= std_lev[-1])
+        i1 = (ds['PRES'].max('N_LEVELS') >= std_lev[-1])
+        dsp = ds.where(i1, drop=True)
+
+        # check if any profile is left
+        if (len(dsp['N_PROF']) == 0):
+            raise ValueError(
```

Comment:
Instead of an error, you could simply return a "None" value and possibly a warning
This could in fact be in the options with an "errors" argument taking "raise" or "ignore" as inputs